### ISSUE TITLE
Can't build on FreeBSD

### ISSUE THREAD

**Issue by jeffpc** [BatchDrake/SuWidgets#16]
I'm not sure what happened, but it was probably a system update at some point that made SuWidgets build fail for me.  Qt hurts my head :)

A long while back I made a simple script I use to build the whole 4-repo combo to get SigDigger built.  The SuWidgets relevant portion:

```
#!/bin/sh

TARGET_DIR=$HOME/bin/dsp

set -e
set -x
...
cd SuWidgets
PKG_CONFIG_PATH=$TARGET_DIR/lib/pkgconfig/ \
QMAKESPEC=/usr/local/share/qt/mkspecs/freebsd-g++ \
	qmake SuWidgetsLib.pro \
	"CONFIG += release" \
	PREFIX=$TARGET_DIR
make
make install
cd ..
...
```

Unfortunately, at some point, the make invocation started failing because it can't find the QDebug header.  The whole SuWidgets build output:

```
+ cd SuWidgets
+ PKG_CONFIG_PATH=/home/jeffpc/bin/dsp/lib/pkgconfig/ QMAKESPEC=/usr/local/share/qt/mkspecs/freebsd-g++ qmake SuWidgetsLib.pro 'CONFIG += release' 'PREFIX=/home/jeffpc/bin/dsp'
SuWidgetsLib.pro:2: Unknown replace function: qtLibraryTarget
SuWidgetsLib.pro:8: Unknown test function: greaterThan
WARNING: Found potential symbol conflict of ColorChooserButton.cpp (ColorChooserButton.cpp) in SOURCES
WARNING: Found potential symbol conflict of ColorChooserButton.h (ColorChooserButton.h) in HEADERS
WARNING: Found potential symbol conflict of FrequencySpinBox.cpp (FrequencySpinBox.cpp) in SOURCES
WARNING: Found potential symbol conflict of FrequencySpinBox.h (FrequencySpinBox.h) in HEADERS
WARNING: Found potential symbol conflict of TimeSpinBox.cpp (TimeSpinBox.cpp) in SOURCES
WARNING: Found potential symbol conflict of TimeSpinBox.h (TimeSpinBox.h) in HEADERS
WARNING: Found potential symbol conflict of MultiToolBox.cpp (MultiToolBox.cpp) in SOURCES
WARNING: Found potential symbol conflict of MultiToolBox.h (MultiToolBox.h) in HEADERS
+ make
make: "/usr/home/jeffpc/src/oss/dsp/SuWidgets/Makefile" line 533: warning: duplicate script for target "moc_ColorChooserButton.cpp" ignored
make: "/usr/home/jeffpc/src/oss/dsp/SuWidgets/Makefile" line 512: warning: using previous script for "moc_ColorChooserButton.cpp" defined here
make: "/usr/home/jeffpc/src/oss/dsp/SuWidgets/Makefile" line 536: warning: duplicate script for target "moc_FrequencySpinBox.cpp" ignored
make: "/usr/home/jeffpc/src/oss/dsp/SuWidgets/Makefile" line 515: warning: using previous script for "moc_FrequencySpinBox.cpp" defined here
make: "/usr/home/jeffpc/src/oss/dsp/SuWidgets/Makefile" line 539: warning: duplicate script for target "moc_TimeSpinBox.cpp" ignored
make: "/usr/home/jeffpc/src/oss/dsp/SuWidgets/Makefile" line 521: warning: using previous script for "moc_TimeSpinBox.cpp" defined here
make: "/usr/home/jeffpc/src/oss/dsp/SuWidgets/Makefile" line 542: warning: duplicate script for target "moc_MultiToolBox.cpp" ignored
make: "/usr/home/jeffpc/src/oss/dsp/SuWidgets/Makefile" line 524: warning: using previous script for "moc_MultiToolBox.cpp" defined here
c++ -c -pipe -Wall -W -O2 -pipe -fstack-protector-strong -fno-strict-aliasing -pthread -D_THREAD_SAFE -fPIC  -DQT_NO_DEBUG -DQT_THREAD_SUPPORT -I/usr/local/share/qt/mkspecs/freebsd-g++ -I. -I../../../../../../local/include -I/include -o ctkRangeSlider.o ctkRangeSlider.cpp
ctkRangeSlider.cpp:22:10: fatal error: 'QDebug' file not found
#include <QDebug>
         ^~~~~~~~
1 error generated.
*** Error code 1

Stop.
```

FWIW, without a QMAKESPEC I get a "QMAKESPEC has not been set, so configuration cannot be deduced."

I'm using the FreeBSD package of qt5 (5.15.2).

**Comment by BatchDrake** [comment]
Look at these lines:

```
SuWidgetsLib.pro:2: Unknown replace function: qtLibraryTarget
SuWidgetsLib.pro:8: Unknown test function: greaterThan
```

I bet this is a Qt4-related version problem. What does `qmake --version` print?

**Comment by jeffpc** [comment]
Heh.  Apparently, FreeBSD has two qmake packages:

```
qmake-3.3.8_2                  The build utility of the Qt 3 project
qt5-qmake-5.15.2p263           Qt Makefile generator (KDE patched)
```
Even though I had only one version of qt libs/includes/etc. installed - qt5, I had the qmake 3 binary.  Installing qt5-qmake (and then using `qmake-qt5` instead of just `qmake`) resulted in a clean build.

So, you were right that it was a qt version mismatch, but were wrong about it being qt4 related ;)

Closing this bug & sorry about the noise.  (I got a SigDigger error about incomplete struct timeval type, but that's for another bug/pull request since it appears completely unrelated.)
